### PR TITLE
Add markDynamoStrictTest

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112768

Add markDynamoStrictTest

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -1302,6 +1299,27 @@ def wrapper(*args, **kwargs):
...
+    fn = cls_or_func
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        torch._dynamo.reset()
+        with unittest.mock.patch("torch._dynamo.config.suppress_errors", False):
+            fn(*args, **kwargs)
+        torch._dynamo.reset()
+    return wrapper
+
+    return wrapper
```
Comment (zou3519) [inline torch/testing/_internal/common_utils.py:1320]: Double return wrapper?

File: torch/testing/_internal/common_utils.py [hunk 2]
```diff
@@ -1302,6 +1299,27 @@ def wrapper(*args, **kwargs):
...
+        cls_or_func.dynamo_strict = True
+        return cls_or_func
+
+    fn = cls_or_func
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        torch._dynamo.reset()
+        with unittest.mock.patch("torch._dynamo.config.suppress_errors", False):
+            fn(*args, **kwargs)
+        torch._dynamo.reset()
```
Comment (zou3519) [inline torch/testing/_internal/common_utils.py:1317]: This doesn't really work on classes, right?

Comment (voznesenskym) [reply torch/testing/_internal/common_utils.py:1317]: the class handling is L1307

File: torch/testing/_internal/common_utils.py (hunk 1 above)
Comment (voznesenskym) [reply torch/testing/_internal/common_utils.py:1320]: ye

File: torch/testing/_internal/common_utils.py (hunk 2 above)
Comment (zou3519) [reply torch/testing/_internal/common_utils.py:1317]: Oh I see, I read this too quickly. Thanks for checking.

**Review by zou3519 — APPROVED** [review]

**Comment by voznesenskym** [comment]
stupid suppress errors is so finicky

File: torch/testing/_internal/common_utils.py [hunk 3]
```diff
@@ -2598,15 +2615,36 @@ def _run_with_retry(self, result=None, num_runs_left=0, report_only=True, num_re
...
-            super_run = torch._dynamo.optimize("aot_eager_decomp_partition")(super_run)
-        elif TEST_WITH_TORCHDYNAMO:
-            # TorchDynamo optimize annotation
-            super_run = torch._dynamo.optimize("eager")(super_run)
-
-        super_run(result=result)
+        test_cls = super_run.__self__
+
+        # Are we compiling?
+        compiled = TEST_WITH_TORCHDYNAMO or TEST_WITH_AOT_EAGER or TEST_WITH_TORCHINDUCTOR
```
Comment (wanchaol) [inline torch/testing/_internal/common_utils.py:2621]: dumb question: how to set these flags in my unit test?

**Pull request closed without merging**